packages/schematics/update/update/npm.ts

```diff
@@ -216,17 +224,36 @@ export function getNpmPackageJson(
             token?: string,
             alwaysAuth?: boolean;
             username?: string;
-            password?: string
+            password?: string;
+            email?: string;
           } = {};
 
           if (alwaysAuth !== undefined) {
             auth.alwaysAuth = alwaysAuth === 'false' ? false : !!alwaysAuth;
           }
 
+          if (email) {
+            auth.email = email;
+          }
+
           if (authToken) {
             auth.token = authToken;
           } else if (token) {
-            auth.token = token;
+            try {
+              // attempt to parse username and password from base64 token
+              // to enable Artifactory / Nexus-like repositories support
+              const parsedToken = Buffer.from(token, 'base64').toString('ascii');
+              const [extractedUsername, extractedPassword] = parsedToken.split(':');
```

Comment:
Looks like it cannot have `:` in the password. Is this expected? You might want to use this instead:

```typescript
const delimiter = ':';
const [extractedUsername, ...passwordArr] = parsedToken.split(delimiter);
const extractedPassword = passwordArr.join(delimiter);
```

Just slightly more ugly, but there's no native function that does it.

Reply:
true, good point. Thanks for reviewing! I will apply the changes soon